src/Zicht/Bundle/UrlBundle/Aliasing/Listener.php

```diff
@@ -147,76 +168,102 @@ protected function isExcluded($url)
      */
     public function onKernelRequest(Event\GetResponseEvent $event)
     {
-        if ($event->getRequestType() === HttpKernelInterface::MASTER_REQUEST) {
-            $request = $event->getRequest();
-            $publicUrl = rawurldecode($request->getRequestUri());
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
 
-            if ($this->isExcluded($publicUrl)) {
-                // don't process urls which are marked as excluded.
-                return;
-            }
+        $request = $event->getRequest();
+        $publicUrl = rawurldecode($request->getRequestUri());
 
-            if ($this->isParamsEnabled) {
-                if (false !== ($queryMark = strpos($publicUrl, '?'))) {
-                    $originalUrl = $publicUrl;
-                    $publicUrl = substr($originalUrl, 0, $queryMark);
-                    $queryString = substr($originalUrl, $queryMark);
-                } else {
-                    $queryString = null;
-                }
+        if ($this->isExcluded($publicUrl)) {
+            // don't process urls which are marked as excluded.
+            return;
+        }
 
-                $parts = explode('/', $publicUrl);
-                $params = array();
-                while (false !== strpos(end($parts), '=')) {
-                    array_push($params, array_pop($parts));
-                }
-                if ($params) {
-                    $publicUrl = join('/', $parts);
+        $queryString = '';
+        if (false !== ($queryMark = strpos($publicUrl, '?'))) {
+            $queryString = substr($publicUrl, $queryMark);
+            $publicUrl = substr($publicUrl, 0, $queryMark);
+        }
 
-                    $parser = new UriParser();
-                    $request->query->add($parser->parseUri(join('/', array_reverse($params))));
+        if ($this->isParamsEnabled) {
+            $parts = explode('/', $publicUrl);
+            $params = array();
```

Comment:
(flavor) is het te overwegen om deze te hernoemen naar `publicUrlParts` en `publicUrlParams`?

Reply:
Kan ja, dit stuk van de code heb ik verder niet aangeraakt. Alleen indent veranderd.